### PR TITLE
Feat: Closing an existing port forward

### LINKED ISSUE #110: Feature Request: Implement Cancel Socket/Port Forwarding
The `Session::request_port_forwarding` [documentation states the following](https://docs.rs/openssh/latest/openssh/struct.Session.html#method.request_port_forward):
> Currently, there is no way of stopping a port forwarding due to the fact that openssh multiplex server/master does not support this.

In ssh, you can cancel port forwarding for multiplexed connections like this:
```
ssh -S [path_to_multiplexing_socket] -O cancel -L/R [forwarding_addresses] 
```
So I don't see why this couldn't be implemented? 
Maybe I'm missing something. Will implement it myself if no one picks up the ticket or tells me why it won't work. :)

### PR DESCRIPTION
**Pull request by jaywonchung**
I implemented port forwarding cancellation for the process mux session implementation.

CC. @NobodyXu 

Closes #110

### CODE REVIEW

**Comment by jonhoo** [comment]
This change is [Reviewable](https://reviewable.io/reviews/openssh-rust/openssh/165)

File: src/native_mux_impl/session.rs [hunk 1]
```diff
@@ -67,6 +67,15 @@ impl Session {
         Ok(())
     }
 
+    pub(crate) async fn cancel_port_forward(
+        &self,
+        forward_type: crate::ForwardType,
+        listen_socket: crate::Socket<'_>,
+        connect_socket: crate::Socket<'_>,
+    ) -> Result<(), Error> {
+        unimplemented!("Port forwarding cancellation is not implemented yet")
```
[the remark this answers is not in the document]
Comment (NobodyXu) [reply src/native_mux_impl/session.rs:76]: openssh-mux-client v0.17.6 has released

**Comment by NobodyXu** [comment]
Sorry I was approving the workflow for your PR, but accidentally hit the "ready for review" butto

**Comment by jaywonchung** [comment]
No worries. But actually good timing you're here -- I'm stuck with testing remote forwarding closure.

Calling `read()` on the socket will actually return a single Line Feed (ascii 10) character. Doing `UnixStream::connect` will actually connect to the socket without any problem, perhaps because there is one more character left to read. I suspected that the `echo` that is used to generate test data from the remote side is appending an extra newline so tried removing the final `\n` from the payload, and now `read()` hangs -- probably because there is not more data left to read and the socket blocks. The code path for local and remote forwarding are exactly the same and I actually verified local forwarding & cancellation working very well in my actual use case. Any ideas?

**Comment by jaywonchung** [comment]
Alright, on my local setup where I do remote forwarding and manually cancel, I found a way. There's one CI failing (`With dependencies at minimal versions`) but I presume that's because this new API is not part of the latest release of `openssh`.

**Review by NobodyXu — CHANGES_REQUESTED** [review]
Sorry I was on another project and didn't check my github

You need to update Cargo.toml, to use latest openssh-mux-client, otherwise that CI would pick the lowest version fitting allowed by version requirements.

**Review by NobodyXu — APPROVED** [review]
Thank you!

**Pull request merged**